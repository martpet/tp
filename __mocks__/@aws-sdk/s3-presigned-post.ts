export const createPresignedPost = vi.fn().mockResolvedValue('dummyPresignedPost');
