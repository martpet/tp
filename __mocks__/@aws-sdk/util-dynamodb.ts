export const unmarshall = vi
  .fn()
  .mockName('unmarshall')
  .mockReturnValue('dummyUnmarshalledItems');
