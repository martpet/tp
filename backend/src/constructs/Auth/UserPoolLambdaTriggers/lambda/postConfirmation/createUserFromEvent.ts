import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
import { DynamoDBDocumentClient, PutCommand } from '@aws-sdk/lib-dynamodb';
import { PostConfirmationTriggerEvent } from 'aws-lambda';

import { usersTableOptions } from '~/consts';

import { getUserPropsFromCognitoEvent } from '../getUserPropsFromCognitoEvent';

const marshallOptions = { removeUndefinedValues: true };
const ddbClient = new DynamoDBClient({});
const ddbDocClient = DynamoDBDocumentClient.from(ddbClient, { marshallOptions });

export const createUserFromEvent = (event: PostConfirmationTriggerEvent) => {
  const userProps = getUserPropsFromCognitoEvent(event);

  const putCommand = new PutCommand({
    TableName: usersTableOptions.tableName,
    Item: userProps,
  });

  return ddbDocClient.send(putCommand);
};
