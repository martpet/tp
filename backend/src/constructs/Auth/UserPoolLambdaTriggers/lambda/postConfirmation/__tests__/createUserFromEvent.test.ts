import { DynamoDBDocumentClient, PutCommand } from '@aws-sdk/lib-dynamodb';
import { mockClient } from 'aws-sdk-client-mock';

import { createUserFromEvent } from '../createUserFromEvent';
import event from './__fixtures__/postConfirmationEvent';

vi.mock('../../getUserPropsFromCognitoEvent');

const ddbMock = mockClient(DynamoDBDocumentClient);

const args = [event] as const;

beforeEach(() => {
  ddbMock.reset();

  ddbMock
    .on(PutCommand)
    .resolves({ Attributes: { dummyPutCommandAttrKey: 'dummyPutCommandAttrValue' } });
});

describe('createUserFromEvent', () => {
  it('sends "PutCommand" to DynamoDB with correct args', async () => {
    await createUserFromEvent(...args);
    expect(ddbMock.commandCalls(PutCommand)[0].args[0].input).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    return expect(createUserFromEvent(...args)).resolves.toMatchSnapshot();
  });
});
