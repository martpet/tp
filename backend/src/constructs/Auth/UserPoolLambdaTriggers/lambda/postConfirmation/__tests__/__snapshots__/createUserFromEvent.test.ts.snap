// Vitest Snapshot v1

exports[`createUserFromEvent > resolves with a correct value 1`] = `
{
  "Attributes": {
    "dummyPutCommandAttrKey": "dummyPutCommandAttrValue",
  },
}
`;

exports[`createUserFromEvent > sends "PutCommand" to DynamoDB with correct args 1`] = `
{
  "Item": {
    "dummyUserPropFromCognitoEventKey": "dummyUserPropFromCognitoEventValue",
    "id": "dummyId",
  },
  "TableName": "TripPics-Users",
}
`;
